File: src/test/java/de/pbauerochse/youtrack/timerangeprovider/TimerangeProviderFactoryTest.java
package de.pbauerochse.youtrack.timerangeprovider;

import de.pbauerochse.youtrack.domain.ReportTimerange;
import de.pbauerochse.youtrack.domain.TimerangeProvider;
import de.pbauerochse.youtrack.domain.timerangeprovider.TimerangeProviderFactory;
import org.junit.Assert;
import org.junit.Test;

import java.time.LocalDate;

/**
 * @author Patrick Bauerochse
 * @since 07.07.15
 */
public class TimerangeProviderFactoryTest {

    @Test
    public void performTest() {

        LocalDate now = LocalDate.now();

        for (ReportTimerange reportTimerange : ReportTimerange.values()) {
            TimerangeProvider timerangeProvider = TimerangeProviderFactory.getTimerangeProvider(reportTimerange, now, now);
            Assert.assertNotNull("TimerangeProvider for timerange " + reportTimerange.name() + " was null", timerangeProvider);

            if (reportTimerange == ReportTimerange.CUSTOM) {
                Assert.assertEquals(now, timerangeProvider.getStartDate());
                Assert.assertEquals(now, timerangeProvider.getEndDate());
            }
        }
    }
}

File: application/src/test/java/de/pbauerochse/worklogviewer/timerangeprovider/TimerangeProviderFactoryTest.kt
package de.pbauerochse.worklogviewer.timerangeprovider

import de.pbauerochse.worklogviewer.domain.ReportTimerange
import de.pbauerochse.worklogviewer.domain.timerangeprovider.TimerangeProviderFactory
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import java.time.LocalDate

internal class TimerangeProviderFactoryTest {

    @Test
    fun performTest() {

        val now = LocalDate.now()

        for (reportTimerange in ReportTimerange.values()) {
            val timerangeProvider = TimerangeProviderFactory.getTimerangeProvider(reportTimerange, now, now)
            assertNotNull(timerangeProvider, "TimerangeProvider for timerange ${reportTimerange.name} was null")

            if (reportTimerange == ReportTimerange.CUSTOM) {
                assertEquals(now, timerangeProvider.timeRange.start)
                assertEquals(now, timerangeProvider.timeRange.end)
            }
        }
    }
}
